### PR TITLE
Microsoft Azure extension pack - Part 2 - Cluster support based on Event Hubs

### PR DESCRIPTION
**Pull request by kaizimmerm**
Leverages [Azure Event Hubs for Apache Kafka](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-for-kafka-ecosystem-overview)  support for communication between cluster nodes. In addition I provided a rudimentary guide to deploy a cluster into [Azure Kubernetes Service (AKS)](https://azure.microsoft.com/en-us/services/kubernetes-service/).

I do admit that these guides should go to the docs section in the homepage. However, I want to keep them here (in this PR) until the Boot 2 migration is complete.

Note: I had to upgrade to Spring Boot 2 in order to get the newest version of [Spring Cloud Stream](https://cloud.spring.io/spring-cloud-stream/) for the Kafka 1.0 support as required by Azure. So this PR depends on https://github.com/eclipse/hawkbit/pull/721

Boot 2 migration for the other modules needs to be done as well:. Here a quick status:
* AWS S3 module: seems to work (to be verified by some of the commiters that use AWS
* Azure modules: works
* MongoDB: needs migration. Looks a bit tricky. A few mandatory calls are not possible anymore with the new Spring data APIs.
* UAA: big breaks here on Spring side. I vote for letting this one go out of maintenance. None of the commiters uses the module. I can offer to provide a reference [Azure Active Directory OAuth2](https://github.com/Microsoft/azure-spring-boot/tree/master/azure-spring-boot-starters/azure-active-directory-spring-boot-starter) extension. Something I can actually maintain.

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by kaizimmerm** [comment]
CQs are all "proceed with checkin" (see list that is part of the PR)

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
    <parent>
       <groupId>org.eclipse.hawkbit</groupId>
       <artifactId>hawkbit-parent</artifactId>
-      <version>0.2.4</version>
+      <version>0.3.0-SNAPSHOT</version>
```
Comment (schabdo) [inline pom.xml]: Shouldn't you wait for 0.3.0M1?

Comment (kaizimmerm) [reply pom.xml]: Sure, as soon as I get Nexus to cooperate

**Pull request merged**